sdks/python/apache_beam/examples/streaming_wordcount_debugging.py

```diff
@@ -0,0 +1,179 @@
+
+  # We use the save_main_session option because one or more DoFn's in this
+  # workflow rely on global context (e.g., a module imported at module level).
+  pipeline_options = PipelineOptions(pipeline_args)
+  pipeline_options.view_as(SetupOptions).save_main_session = True
+  pipeline_options.view_as(StandardOptions).streaming = True
+  p = beam.Pipeline(options=pipeline_options)
+
+  # Read from PubSub into a PCollection.
+  if known_args.input_subscription:
+    lines = p | beam.io.ReadStringsFromPubSub(
+        subscription=known_args.input_subscription)
+  else:
+    lines = p | beam.io.ReadStringsFromPubSub(topic=known_args.input_topic)
+
+  # Count the occurrences of each word.
+  def count_ones(word_ones):
+    (word, ones) = word_ones
+    return (word, sum(ones))
+
+  counts = (lines
+            | 'AddTimestampFn' >> beam.ParDo(AddTimestampFn())
+            | 'After AddTimestampFn' >> ParDo(PrintFn('After AddTimestampFn'))
+            | 'Split' >> (beam.ParDo(WordExtractingDoFn())
+                          .with_output_types(six.text_type))
+            | 'PairWithOne' >> beam.Map(lambda x: (x, 1))
+            | beam.WindowInto(window.FixedWindows(5, 0))
+            | 'AfterWindow' >> ParDo(PrintFn('AfterWindow'))
+            | 'GroupByKey' >> beam.GroupByKey()
+            | 'After GroupByKey' >> ParDo(PrintFn('After GroupByKey'))
+            | 'CountOnes' >> beam.Map(count_ones))
+
+  # Format the counts into a PCollection of strings.
+  def format_result(word_count):
+    (word, count) = word_count
+    return '%s: %d' % (word, count)
+
+  output = counts | 'format' >> beam.Map(format_result)
+
+  # Write to PubSub.
+  # pylint: disable=expression-not-assigned
+  output | beam.io.WriteStringsToPubSub(known_args.output_topic)
+
+  def check_gbk_format():
+    # A matcher that checks that the output of GBK is of the form word: count.
+    def matcher(elements):
+      # pylint: disable=unused-variable
+      actual_elements_in_window, window = elements
+      for elm in actual_elements_in_window:
+        assert re.match(r'\S+:\s+\d+', elm) is not None
+    return matcher
+
+  # Check that the format of the output is correct.
+  assert_that(
+      output,
+      check_gbk_format(),
+      use_global_window=False,
+      label='Assert word:count format.')
+
+  # Check also that elements are ouput in the right window.
+  # This expects exactly 1 occurrence of any subset of the elements
+  # 150, 151, 152, 153, 154 in the window [150, 155)
+  # or exactly 1 occurrence of any subset of the elements
+  # 210, 211, 212, 213, 214 in the window [210, 215).
+  expected_window_to_elements = {
+      window.IntervalWindow(150, 155): [
+          ('150: 1'), ('151: 1'), ('152: 1'), ('153: 1'), ('154: 1'),
+      ],
+      window.IntervalWindow(210, 215): [
+          ('210: 1'), ('211: 1'), ('212: 1'), ('213: 1'), ('214: 1'),
+      ],
+  }
+
+  # To make it pass, publish numbers in [150-155) or [210-215) with no repeats.
```

Comment:
This really needs to move to the top of the file. A user needs to know how to execute this pipeline, and what kind inputs it needs as part of the top level comment.

Reply:
What to give to make it pass or fail pertains only `expected_window_to_elements`.
However, I have added a usage section to the top of the file, like debugging_wordcount.py